File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participants, String[] completions) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String participant : participants) {
            map.put(participant, map.getOrDefault(participant, 0) + 1);
        }
        
        for (String completion : completions) {
            map.put(completion, map.get(completion) - 1);
        }
        
        for (String participant : participants) {
            if (map.get(participant) > 0) {
                return participant;
            }
        }
        
        return "";  
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let map = new Map()
    
    for(let people of participant) {
        map.set(people, (map.get(people) || 0) + 1);
    }
    
    for(let people of completion) {
        if(map.get(people)) {
            map.set(people, map.get(people) -1);
        }
    }
    
    for (let [key, value] of map) {
        if (value > 0) return key;
    }
    
}